common/retry.py

```diff
@@ -41,6 +39,8 @@ def wrap(  # pylint: disable=too-many-arguments
         exception_type=Exception,
         retry_on_false=False):
     """Retry decorator for a function."""
+    # To avoid circular dependency.
```

Comment:
This seems to be hiding the dependency rather than avoiding it.
It's fine if this is needed because this CL didn't introduce this problem but comment is misleading IMO.

Reply:
Yes otherwise it breaks CI. Since we need to use logs in retry, i dont see any other way to break the circular deps. This is hacky, but no other way.